server/src/main/java/org/opensearch/action/support/replication/ReplicationOperation.java

```diff
@@ -118,6 +120,22 @@ public ReplicationOperation(
         long primaryTerm,
         TimeValue initialRetryBackoffBound,
         TimeValue retryTimeout
+    ) {
+        this(request, primary, listener, replicas, logger, threadPool, opType, primaryTerm, initialRetryBackoffBound, retryTimeout, false);
+    }
+
+    public ReplicationOperation(
+        Request request,
+        Primary<Request, ReplicaRequest, PrimaryResultT> primary,
+        ActionListener<PrimaryResultT> listener,
+        Replicas<ReplicaRequest> replicas,
+        Logger logger,
+        ThreadPool threadPool,
+        String opType,
+        long primaryTerm,
+        TimeValue initialRetryBackoffBound,
+        TimeValue retryTimeout,
+        boolean forceReplicationIfRemoteTranslogEnabled
```

Comment:
Inject the proxy implementation?

Reply:
same comment as the similar comment about injecting -

@Bukhtawar - this would be difficult since the PerformOnReplicaProxy implementation changes per ShardRouting depending on TransportReplicationMode (need for replication) i.e. the replication strategy is not universal to all ShardRoutings per replication request.

ReplicationOperation would lead to a universal strategy for the request which is above the granularity at which we need PerformOnReplicaProxy.

Reply:
What if your proxy takes ReplicationAwareShardRouting as a param?  Or add a force property to ReplicationRequest and have a general TranslogReplicationProxy that blocks based on the passed in ShardRouting?

Reply:
Not sure if I got you completely. But let me attempt to have a common `TranslogReplicationProxy` which can have a input regarding the force property from underlying TransportReplicationAction implementation and uses ReplicationAwareShardRouting to decide the behaviour of replication.

Reply:
@Bukhtawar  injection done in ctor itself. pls take a look
@mch2 did as you mentioned, pls take a look